### PR TITLE
Create Kynea_numbers.dart

### PR DESCRIPTION
**Pull request by RimjhimGupta**
Kynea number is a positive integer of the form:  4^n + 2^(n+1)- 1
This pull request is for Hacktober Fest 2020.

# Welcome to Dart community

### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/Kynea_numbers.dart [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// CPP code to find nth Kynea number
+
+//#include <bits/stdc++.h>
+//using namespace std;
```
Comment (Parowicz) [inline maths/Kynea_numbers.dart:4]: That's unexpected!

File: maths/Kynea_numbers.dart [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  // print nth kynea number
+  print(nthKyneaNumber(n));
+
+  return 0;
+}
+/*void main() {
+  for (int i = 0; i < 5; i++) {
+    print('hello ${i + 1}');
+  }
+}*/
```
Comment (Parowicz) [inline maths/Kynea_numbers.dart:37]:
```suggestion
```

**Review by Parowicz — CHANGES_REQUESTED** [review]

File: maths/Kynea_numbers.dart (hunk 1 above)
Comment (RimjhimGupta) [reply maths/Kynea_numbers.dart:4]: Sir, I am aware of C++ and learning Dart. I won't repeat this mistake in future.
Sorry Sir

File: maths/Kynea_numbers.dart [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// Function to calculate nth kynea number
```
Comment (Parowicz) [inline maths/Kynea_numbers.dart:1]: You could add some tests
```suggestion
import 'package:test/test.dart';

// Function to calculate nth kynea number
```

File: maths/Kynea_numbers.dart [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+// Driver Program
+int main() {
+  int n = 8;
+
+  // print nth kynea number
+  print(nthKyneaNumber(n));
+
+  return 0;
+}
```
Comment (Parowicz) [inline maths/Kynea_numbers.dart:27]: For example
```suggestion

void main() {
  test("1th Kynea number equals to 7", () {
    expect(nthKyneaNumber(1), equals(7));
  });

  test("4th Kynea number equals to 287", () {
    expect(nthKyneaNumber(4), equals(287));
  });

  test("6th Kynea number equals to 4223", () {
    expect(nthKyneaNumber(6), equals(4223));
  });

  test("10th Kynea number equals to 1050623", () {
    expect(nthKyneaNumber(10), equals(1050623));
  });
}
```

**Review by Parowicz — CHANGES_REQUESTED** [review]

**Comment by RimjhimGupta** [comment]
@Parowicz Sir, I request you to please merge my pull request for Hacktober Fest 2020.

**Review by Parowicz — APPROVED** [review]
Looks Great! Thank You!

**Comment by RimjhimGupta** [comment]
@Parowicz Thank you Sir

**Pull request merged**